torch/csrc/jit/runtime/graph_executor.cpp

```diff
@@ -736,7 +736,13 @@ c10::intrusive_ptr<Future> GraphExecutor::runAsync(Stack& stack) {
 }
 
 size_t GraphExecutor::getDefaultNumBailOuts() {
-  return getProfilingMode() ? getBailoutDepth().load() : 0;
```

Comment:
Could we consider not even querying `GraphExecutor` if `! getGraphExecutorOptimize()` ?

Reply:
Or dispatching to the SimpleExecutor if `!getGraphExecutorOptimize()`

Reply:
From looking around maybe there's not a great place to do this... LGTM

Reply:
That was my impression as well tbh